### DISCUSSION TITLE
[ITA-SSO_FRONT] Discussion on Initial Development and Configuration of sso_front (#799)

### DISCUSSION THREAD

**Discussion by missvader** [IT-Academy-BCN/ita-wiki#803] [Q&A]
**Description:**

Hello team,
This discussion focuses on the initiation of the development for the sso_front component in relation to issue #799, titled "[FE] Start with sso_front".

**Objective:**

- Establish an initial structure for the files and folders of the  `sso_front`  project.
- Set up Continuous Integration (CI) to ensure automated building and testing of the code.

**Proposed Steps:**

1. Create a basic structure of folders and files for the `sso_front` project.
2. Initiate the CI setup to ensure automated building and testing of the code.
3. Iterate through any challenges and issues encountered during the development process.

**Comment by octapf** [comment]
Breaking the ice!

I have a question. 

- How do we start testing the structure?

I remember Kevin saying it would be nice if each of us tries to do it on its own

**Reply by marmorrei** [reply]
I agree we all have to understand what the structure is and how it works but, do we have to do repeated work? Or instead agree on what config will do each one of us and then share it with the rest? I am speaking from total ignorance...

**Reply by missvader** [reply]
I believe that precisely because we are a team, we should organize ourselves a bit.
Perhaps, initially, we should look for some documentation on SSO in the front end and how to implement it. 
Then create a basic outline of the structure we need and the steps to follow. 
And from there, we can divide the work into tasks

**Comment by missvader** [comment]
My idea is for us to understand what we need and what we should do. I believe that a visual schema of the structure would be beneficial for all of us.
And I think that repeated work is not a good idea.
Just my opinion

**Reply by marmorrei** [reply]
Do you mean to do something like this and we can see, to begin with, what are we facing before doing next steps?
![image](https://github.com/IT-Academy-BCN/ita-wiki/assets/124587546/5c91491c-5189-4b55-8fbf-d10b67da39e4)

**Reply by missvader** [reply]
![image](https://github.com/IT-Academy-BCN/ita-wiki/assets/94196643/f6b30dfa-0ce5-489c-9e78-437e31272f48)
We need to set up a new application, a front end, following the same structure as we have in ita-wiki. During the configuration phase, we have to link it to the existing SSO.

**Reply by missvader** [reply]
For now, inside src folder, we need all these empty folders and a `vite-env.d.ts` file 
![image](https://github.com/IT-Academy-BCN/ita-wiki/assets/94196643/5868dcac-349c-4d75-8606-11450ed2ba66)

**Reply by marmorrei** [reply]
Do you think it is best to create all the folders inside `src` first and after that initialise the react-vite project or the other way around?

**Reply by missvader** [reply]
I suppose initializing first, as that will already create many of the folders we need. What do you think?

**Reply by marmorrei** [reply]
Yes, I think so, because anyway it will have to be done, so we use whatever vite create for us.

**Comment by missvader** [comment]
Assuming everyone is on board with this approach, we could break down the task into subtasks:

1.     Create the initial structure of `sso_front`, empty.
2.     Configure `package.json`, `tsconfig`, `dockerfile`, `eslint`, `prettier`, and `vite.config`. (Here, we need to set up a proxy to connect with 
    SSO)
3.     Configure tests in the `src` folder.
4.     Implement coverage and integrate with SonarCloud.

This could serve as a task checklist.
And I believe the best way to approach this would be by creating a main branch for the issue, and from there, create sub-branches for each of the tasks.

**Reply by marmorrei** [reply]
Ok! If we write a comment on the issue #799 with your checklist as checklist items, each item can be converted to issue, and from there open the corresponding sub-branch.

**Reply by missvader** [reply]
ok. Do it

**Reply by marmorrei** [reply]
Done!

**Reply by marmorrei** [reply]
I've just realised that we have to modify the names of the issues to match naming conventions... can we do that on the issue itselt??

**Reply by marmorrei** [reply]
I guess it has to be with [FE] at the beginning?? or [FE] [SSO]?

**Reply by missvader** [reply]
You can edit the issue itself. Maybe [FE] [SSO] ?

**Reply by missvader** [reply]
I have created a main branch where we can submit pull requests for branches related to different tasks/issues.

**Reply by marmorrei** [reply]
I have modified the [FE][SSO] + labels and project inside each issue

**Comment by marmorrei** [comment]
Hello again guys!

As Kevin mentioned during the daily, he wants the PR of the final configuration to be able to see that is working.
That means we need to finish the last 2 list items from the checklist that @missvader did (#799) and whatever else is needed. I will assign one to myself to see what I can do. 

Also, he doesn't want to see the pr's from the sub-branches to the branch of the issue #799. If I am not mistaken, I think he said yesterday for everyone working on this issue, to commit to the branch of the issue #799 and then, when doing PR to 'main', squash all the commits. Do we continue in this way then?

And one last thing: do we merge directly the PR #818 ??

**Reply by missvader** [reply]
I think it’s about merging the pull requests of the tasks into the main issue branch ourselves, and when the main issue is ready, then we request a review

**Reply by marmorrei** [reply]
Ok, then we continue same way, but merging directly the sub-branches.
